backend/src/GraphQl/OpenIdConnect/Application/ApplicationType.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using HotChocolate;
+using HotChocolate.Types;
+using Metabase.Authorization;
+using Metabase.Data;
+using Metabase.GraphQl.Users;
+using Microsoft.AspNetCore.Identity;
+using OpenIddict.Core;
+
+namespace Metabase.GraphQl.OpenIdConnect.Application;
+
+public sealed class ApplicationType
+    : ObjectType<OpenIdApplication>
+{
+    protected override void Configure(
+        IObjectTypeDescriptor<OpenIdApplication> descriptor
+    )
+    {
+        const string suffixedName = nameof(ApplicationType);
+        descriptor.Name(suffixedName.Remove(suffixedName.Length - "Type".Length));
+
+        descriptor.Field(t => t.Permissions).Type<NonNullType<ListType<StringType>>>().Resolve(context =>
+        {
+            var application = context.Parent<OpenIdApplication>();
+            var permissions = JsonSerializer.Deserialize<List<string>>(application.Permissions!);
+            return permissions?.FindAll(permission => permission.Contains("api")).ToList();
+        });
+
+        descriptor.Field(t => t.RedirectUris).Type<NonNullType<StringType>>().Resolve(context =>
+        {
+            var application = context.Parent<OpenIdApplication>();
+            if (string.IsNullOrEmpty(application.RedirectUris))
+            {
+                return string.Empty;
```

Comment:
Returning an empty string is error prone because it is obviously not a functioning redirect URI. The options that come up to me are:

* If there must **not** be a redirect URI, then the field should be nullable and we should just return `null` here.
* If there must be a redirect URI, then this case is an application error and we should throw a `GraphQLException` here.

The same is true for the other fields below.